src/table/src/engine/manager.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
+use crate::engine::TableEngineRef;
+
+pub trait TableEngineManager: Send + Sync {
+    fn engine(&self, name: &str) -> Option<TableEngineRef>;
+    fn register_engine(&self, name: &str, engine: TableEngineRef);
+    fn default(&self) -> TableEngineRef;
```

Comment:
I don't think we need this `default` method here. Just treat "mito" table engine as a regular one, and register it when TableEngineManager is created.